src/cascadia/TerminalSettingsEditor/Actions.h

```diff
@@ -59,20 +62,29 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         void AttemptAcceptChanges(hstring newKeyChordText);
         void DeleteKeyBinding() { _DeleteKeyBindingRequestedHandlers(*this, _Keys); }
 
-        VIEW_MODEL_OBSERVABLE_PROPERTY(bool, IsInEditMode, false);
+        // ProposedAction:   the entry selected by the combo box; may disagree with the settings model.
+        // CurrentAction:    the combo box item that maps to the settings model value.
+        // AvailableActions: the list of options in the combo box; both actions above must be in this list.
+        VIEW_MODEL_OBSERVABLE_PROPERTY(IInspectable, ProposedAction);

```

Comment:
Why `IInspectable` here and `Object` on the interface? The only sets and gets I see for this are from hstrings anyway. Why not just use the `hstring`? Does this swap with another type somewhere that's not obvious?

Reply:
This one is dumb. I even asked Leon about it a while back and we couldn't figure it out. `SelectedItem` takes an `IInspectable`, but for some reason XAML/WinRT isn't smart enough to automatically convert between an `hstring` and an `IInspectable`:

From Actions.xaml.g.hpp:
```cpp
            case 5: // Actions.xaml line 205
                {
                    auto targetElement = target.as<::winrt::Windows::UI::Xaml::Controls::ComboBox>();
                    obj5 = targetElement;
                    obj5.RegisterPropertyChangedCallback(::winrt::Windows::UI::Xaml::Controls::Primitives::Selector::SelectedItemProperty(),
                        [this] (DependencyObject const& sender, DependencyProperty const& prop)
                        {
                            if (IsInitialized())
                            {
                                // Update Two Way binding
                                if (GetDataRoot() != nullptr)
                                {
                                    // LOOK HERE! winrt:to_hstring won't compile! :(
                                    GetDataRoot().ProposedAction(::winrt::to_hstring(obj5.SelectedItem()));
                                }
                            }
                        });
                }
                break;
```